### PR TITLE
feat: Start/stop performance metrics collection for each test case without glue code

### PR DESCRIPTION
**Pull request by PreethiMaai**

### CODE REVIEW

File: cypress/support/cypress-support/src/main.js [hunk 1]
```diff
@@ -89,15 +89,6 @@ export default function (module) {
...
     cy.getModuleReqIdJson();
-    if (UTILS.getEnvVariable(CONSTANTS.PERFORMANCE_METRICS) == true) {
-      cy.startOrStopPerformanceService(CONSTANTS.INITIATED).then((response) => {
-        if (response) {
-          Cypress.env(CONSTANTS.IS_PERFORMANCE_METRICS_ENABLED, true);
-        }
-      });
-    } else {
-      cy.log(CONSTANTS.PERFORMANCE_METRICS_NOT_ACTIVE);
-    }
```
Comment (neeradanelxsi) [inline cypress/support/cypress-support/src/main.js:100]: why did you remove this code ? This function is to start/stop metrics collection and that should reside in before operation

File: cypress/support/cypress-support/src/main.js [hunk 2]
```diff
@@ -242,6 +261,17 @@ export default function (module) {
...
+    if (UTILS.getEnvVariable(CONSTANTS.IS_PERFORMANCE_METRICS_ENABLED, false) == true) {
+      cy.startOrStopPerformanceService(CONSTANTS.STOPPED).then((response) => {
+        if (response) {
+          Cypress.env(CONSTANTS.IS_PERFORMANCE_METRICS_ENABLED, false);
+        }
+      });
+    }
+  });
+
   /**
```
Comment (neeradanelxsi) [inline cypress/support/cypress-support/src/main.js:275]: This is not right, why are we stopping metrics collection in aftereach

File: cypress/support/cypress-support/src/main.js [hunk 3]
```diff
@@ -132,8 +123,36 @@ export default function (module) {
   // beforeEach
   beforeEach(() => {
     UTILS.getEnvVariable(CONSTANTS.FB_INTERACTIONLOGS).clearLogs();
-    cy.getBeforeOperationObject();
-    UTILS.destroyGlobalObjects([CONSTANTS.LIFECYCLE_APP_OBJECT_LIST]);
```
Comment (neeradanelxsi) [inline cypress/support/cypress-support/src/main.js:136]: why are we adding a .then for lines 135 and 136. 
You can create a new command which internally creates marker and save start time there. Then that command can be used in beforeeach right

File: cypress/support/cypress-support/src/main.js [hunk 4]
```diff
@@ -132,8 +123,36 @@ export default function (module) {
...
+
+          cy.startOrStopPerformanceService(CONSTANTS.INITIATED).then((response) => {
+            if (response) {
+              Cypress.env(CONSTANTS.IS_PERFORMANCE_METRICS_ENABLED, true);
+            }
+          });
+        }
+      } else {
+        cy.log(CONSTANTS.PERFORMANCE_METRICS_NOT_ACTIVE);
+      }
```
Comment (neeradanelxsi) [inline cypress/support/cypress-support/src/main.js:154]: line 146 is nto 150 is not correct

File: cypress/support/cypress-commands/commands.js [hunk 5]
```diff
@@ -1549,3 +1548,23 @@ Cypress.Commands.add('startOrStopInteractionsService', (action) => {
 Cypress.Commands.add('envConfigSetup', () => {
   fireLog.info('No additional config module environment setup');
 });
+
```
Comment (neeradanelxsi) [inline cypress/support/cypress-commands/commands.js:1551]: comments and function descriptions are missing

Comment (PreethiMaai) [reply cypress/support/cypress-commands/commands.js:1551]: added

File: cypress/support/cypress-support/src/main.js (hunk 1 above)
Comment (PreethiMaai) [reply cypress/support/cypress-support/src/main.js:100]: reverted the change

File: cypress/support/cypress-support/src/main.js (hunk 3 above)
Comment (PreethiMaai) [reply cypress/support/cypress-support/src/main.js:136]: created initiatePerformanceMetrics command to save start time and create marker

File: cypress/support/cypress-commands/commands.js [hunk 6]
```diff
@@ -1549,3 +1548,34 @@ Cypress.Commands.add('startOrStopInteractionsService', (action) => {
...
+    // Request to create a marker
+    const requestMap = {
+      method: CONSTANTS.REQUEST_OVERRIDE_CALLS.CREATE_MARKER,
+      params: scenarioName,
+    };
+    fireLog.info(`Firebolt Call to 1st party App: ${JSON.stringify(requestMap)} `);
+    cy.sendMessagetoPlatforms(requestMap).then((result) => {
+      fireLog.isTrue(result.success, 'Response for marker creation: ' + JSON.stringify(result));
+    });
+
```
Comment (neeradanelxsi) [inline cypress/support/cypress-commands/commands.js:1574]: logs 1570 and 1572 to be removed before merge. the log issue will be taken in  https://ccp.sys.comcast.net/browse/FIRECERT-2480

File: cypress/support/validations/performanceValidation.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+ * @module performanceValidation
+ * @function performanceValidation
+ * @description Validates performance metrics based on provided validation parameters.
+ * @param {Object} object - The object to validate
+ * @example
+ * cy.performanceValidation({})
+ */
+Cypress.Commands.add('performanceValidation', (object) => {
+  // Check if performance metrics are enabled
+  if (UTILS.getEnvVariable('performanceMetrics')) {
```
Comment (jbigel) [inline cypress/support/validations/performanceValidation.js:32]: Here: https://github.com/rdkcentral/firebolt-certification-suite/pull/236/files#diff-a0c4b6497a6be8ded74b56b3268804337f0607a41306ae7c236b91b7a7fab118R1564 - use a constant for performance metrics, can we not use the same here?

File: cypress/support/cypress-commands/commands.js [hunk 8]
```diff
@@ -1549,3 +1548,34 @@ Cypress.Commands.add('startOrStopInteractionsService', (action) => {
...
+/**
+ * @module commands
+ * @function initiatePerformanceMetrics
+ * @description Creates a marker and saves the start time in THRESHOLD_MONITOR_START_TIME env, if performance metrics is enabled.
+ * @example
+ * cy.initiatePerformanceMetrics()
+ */
+Cypress.Commands.add('initiatePerformanceMetrics', () => {
+  // Retrieve the scenario name from the env.
+  const scenarioName = Cypress.env(CONSTANTS.SCENARIO_NAME);
```
[the remark this answers is not in the document]
Comment (PreethiMaai) [reply cypress/support/cypress-commands/commands.js]: defined scenarioName inside if block.

File: cypress/support/validations/performanceValidation.js (hunk 7 above)
Comment (PreethiMaai) [reply cypress/support/validations/performanceValidation.js]: changed to constant

**Review by neeradanelxsi — APPROVED** [review]

**Review by jbigel — APPROVED** [review]

File: cypress/support/validations/performanceValidation.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+/**
+ * @module performanceValidation
+ * @function performanceValidation
+ * @description Validates performance metrics based on provided validation parameters.
+ * @param {Object} object - The object to validate
+ * @example
+ * cy.performanceValidation({})
+ */
+Cypress.Commands.add('performanceValidation', (object) => {
```
Comment (kevinshahfws) [inline cypress/support/validations/performanceValidation.js:30]: I am little confused here.. Is this glue to start performance metrics or validate?

**Review by kevinshahfws — CHANGES_REQUESTED** [review]

File: cypress/support/validations/performanceValidation.js (hunk 9 above)
Comment (PreethiMaai) [reply cypress/support/validations/performanceValidation.js:30]: This function is to validate performance metrics.

**Review by kevinshahfws — APPROVED** [review]
